wouso/games/teamquest/tests.py

```diff
@@ -200,6 +209,152 @@ def test_quest_status_create(self):
+                self.assertEqual(level_status.next_level.level.index - 1, level_status.level.index)
+            else:
+                # If last level, check next_level is none
+                self.assertEqual(level_status.next_level, None)
+
+    def test_level_status_unlocked_questions(self):
+        status = TeamQuestStatus.create(group=self.group, quest=self.quest)
+        total_levels = status.levels.all().count()
+
+        for level_status in status.levels.all():
+            # The first level is a special case, as all the questions are unlocked
+            if level_status.level.index == 1:
+                for question in level_status.questions.all():
+                    self.assertTrue(question in level_status.unlocked_questions)
+
+            else:
+                for question in level_status.questions.all():
+                    # Check if question is not in unlocked_questions
+                    self.assertTrue(question not in level_status.unlocked_questions)
+                    # Unlock current question
+                    question.lock = 'U'
+                    question.save()
+                    # Check if it now is in unlocked_questions
+                    self.assertTrue(question in level_status.unlocked_questions)
+
+    def test_level_status_completed(self):
+        status = TeamQuestStatus.create(group=self.group, quest=self.quest)
+
+        for level_status in status.levels.all():
+            # Check that a level is completed only after all the questions are answered
+            for question in level_status.questions.all():
+                self.assertEqual(level_status.completed, False)
+                question.state = 'A'
+                question.save()
+
+            self.assertEqual(level_status.completed, True)
 
     def test_quest_status_time_finished_before_time_started(self):
         pass
+
+class TeamQuestGameTest(TestCase):
+    def setUp(self):
+        category = Category.add('quest')
+        number_of_levels = 5
+        self.questions = []
+        for index in range(number_of_levels * (number_of_levels + 1) / 2):
+            question = Question.objects.create(text='question'+str(index+1), answer_type='F',
+                                               category=category, active=True)
+            self.questions.append(question)
+            answer = Answer.objects.create(text='answer'+str(index+1), correct=True, question=question)
+
+        self.levels = []
+        # The start index of the questions sequence that goes in a level
+        base = 0
+        for index in range(number_of_levels):
+            level = TeamQuestLevel.create(quest=None, bonus=0,
+                                          questions=self.questions[base:base+number_of_levels-index])
+            self.levels.append(level)
+            base += number_of_levels - index
+
+        self.quest = TeamQuest.create(title="_test_quest", start_time=datetime.datetime.now(),
```

Comment:
You can just use an empty list of levels considering the test written in this class doesn't care about those.